### ISSUE TITLE
Increase mas's minimum supported macOS version from 10.11 to 10.13 or newer?

### ISSUE THREAD

**Issue by rgoldberg** [mas-cli/mas#578]
Should mas's minimum supported macOS version be increased from 10.11 to 10.13 or newer?

The user base on macOS 10.11 & 10.12 is probably vanishingly small.

`.swift-version` has specified since 2022 that mas uses Swift 5.7, but that isn't in any released mas version. The code already uses the [if let shorthand](https://github.com/swiftlang/swift-evolution/blob/main/proposals/0345-if-let-shorthand.md) from 5.7, and possibly other 5.7 features.

Moreover, `main` has been switched from Commandant to [Swift Argument Parser 1.5.0](https://github.com/apple/swift-argument-parser), which [requires Swift 5.7+](https://github.com/apple/swift-argument-parser?tab=readme-ov-file#supported-versions).

[Swift 5.7 can only be compiled by Xcode 14+. Xcode 14+ only supports macOS deployment targets on 10.13+](https://developer.apple.com/support/xcode/)

Thus, I think that all references to Swift versions in `main` should be set to the newest Swift 5.7.x, and mas should be configured to explicitly only support macOS 10.13+.

We might want to increase the minimum macOS version to 10.15 when we release 2.0.0, which will allow the use of [Swift Concurrency](https://docs.swift.org/swift-book/documentation/the-swift-programming-language/concurrency/) & [swift-testing](https://github.com/swiftlang/swift-testing) or newer versions of Nimble & Quick. Other newer macOS versions wouldn't unlock as many new useful APIs, and might orphan too many users.

The benefits of which I'm aware of increasing the minimum macOS version for running mas are (by minimum version):

- 10.12
  - `ISO8601DateFormatter`
- 10.13
  - Swift 5.7+
  - PromiseKit 8+ (currently 8.1.2)
  - `JSONSerialization.WritingOptions.sortedKeys`
- 10.15
  - [Concurrency](https://docs.swift.org/swift-book/documentation/the-swift-programming-language/concurrency/)
  - [swift-testing](https://github.com/swiftlang/swift-testing)
  - Nimble 11+ (currently 13.x)
  - Quick 6+ (currently 7.x)
  - `JSONSerialization.WritingOptions.withoutEscapingSlashes`
- 11
  - `Logger`
- 12
  - `Date.ISO8601Format`
  - `Date.ISO8601FormatStyle`
- 13
  - `Regex` & related
  - `Locale.Currency`
- 15
  - `RangeSet`

I know that there are other benefits, I just haven't yet identified them.

**Comment by rgoldberg** [comment]
### TL;DR
mas is supposed to run on macOS 10.11+, as configured in its `Package.swift`.

Xcode never complained about the following potential version mismatch, so mas from `main` might not be able to run on macOS 10.11 or 10.12.

`main` has switched from Commandant to Swift Argument Parser (SAP) 1.5.0.

I just saw that [SAP 1.3.0+ requires Swift 5.7+](https://github.com/apple/swift-argument-parser?tab=readme-ov-file#supported-versions) (I didn't know that I had to search manually search for version compatibility as I had expected Xcode to complain about any version mismatches since config files like `Package.swift` specify required versions).

`Package.swift` in `main` configured mas to use Swift 5.6.1, as [that is the newest Swift supported by any Xcode version that includes macOS 10.11 & 10.12 amongst its deployment targets (Xcode 13.4 is the newest)](https://developer.apple.com/support/xcode/).

Swift 5.7+ requires Xcode 14+, but Xcode 14+ only supports deployment targets of macOS 10.13+ (see the prior link).

SAP can thus only target macOS 10.13+, leaving macOS 10.11 & 10.12 unsupported.

**Comment by rgoldberg** [comment]
@phatblat mas is supposed to run on macOS 10.11+, as configured in its `Package.swift`.

Xcode never complained about the following potential version mismatch, so mas from `main` might not be able to run on macOS 10.11 or 10.12.

`main` has switched from Commandant to Swift Argument Parser (SAP) 1.5.0.

I just saw that [SAP 1.3.0+ requires Swift 5.7+](https://github.com/apple/swift-argument-parser?tab=readme-ov-file#supported-versions) (I didn't know that I had to search manually search for version compatibility as I had expected Xcode to complain about any version mismatches since config files like `Package.swift` specify required versions).

`Package.swift` in `main` configured mas to use Swift 5.6.1, as [that is the newest Swift supported by any Xcode version that includes macOS 10.11 & 10.12 amongst its deployment targets (Xcode 13.4 is the newest)](https://developer.apple.com/support/xcode/).

Swift 5.7+ requires Xcode 14+, but Xcode 14+ only supports deployment targets of macOS 10.13+ (see the prior link).

SAP can thus only target macOS 10.13+, leaving macOS 10.11 & 10.12 unsupported.
